internal/yggdrasil/yggdrasil.go

```diff
@@ -416,3 +429,114 @@ func (h *Handler) setStorageConfiguration(ctx context.Context,
 
 	return err
 }
+
+func (h *Handler) createSecretList(ctx context.Context, logger logr.Logger, deployments []v1alpha1.EdgeDeployment, device *v1alpha1.EdgeDevice) (models.SecretList, error) {
+	list := models.SecretList{}
+
+	// create map of secret names and keys
+	secretMap := secretMapType{}
+	for _, deployment := range deployments {
+		podSpec := deployment.Spec.Pod.Spec
+		allContainers := append(podSpec.InitContainers, podSpec.Containers...)
+		for _, container := range allContainers {
+			extractSecretsFromContainer(&container, secretMap)
+		}
+	}
+	fmt.Println(secretMap)
+
+	// read secrets
+	secretObjs := []corev1.Secret{}
+	for name, keys := range secretMap {
+		optional := keys == nil
+		secretObj := corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: name, Namespace: device.Namespace}}
+		err := h.client.Get(ctx, client.ObjectKeyFromObject(&secretObj), &secretObj)
+		if err != nil {
+			if errors.IsNotFound(err) && optional {
+				continue
+			}
+			return nil, err
+		}
+		for key := range keys {
+			if secretObj.Data != nil {
+				if _, ok := secretObj.Data[key]; ok {
+					continue
+				}
+			}
+			return nil, fmt.Errorf("missing secret key. secret: %s. key: %s", name, key)
```

Comment:
I wonder why we stop iterating over the secrets? What do you think about `warn` with descriptive message and continue iterating?

Reply:
I followed the same method as with workloads/deployments. I do agree that there is room to re-think the way we handle errors.

Reply:
also...if we do not stop here then it will fail in the device. Since EdgeDeployment is used by many devices it means that many devices will have failures.